File: hashtable/36_valid_sudoku.java
package hashtable;

// https://leetcode.com/problems/valid-sudoku/

import java.util.*;

class Solution {
    public boolean isValidSudoku(char[][] board) {

        HashSet<String> set = new HashSet<String>();

        for (int i = 0; i < board.length; i++){
            for (int j = 0; j < board[0].length; j++){
                char c = board[i][j];
                if (!set.add(c + "row" + i) ||
                    !set.add(c + "col" + j) ||
                    !set.add(c + "sub" + i/3 + " " + j/3))
                    return false;
            }
        }
        return true;
        
    }
}

File: hash_table/36_valid_sudoku.py
# https://leetcode.com/problems/valid-sudoku/

from typing import List

class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:

        res = []
        for i in range(len(board)):
            for j in range(len(board[0])):
                curr = board[i][j]
                if curr != ".":
                    res += [(i, curr), (curr, j), (i // 3, j // 3, curr)]
        return len(res) == len(set(res))